packages/api/src/submittable/Extrinsic.ts

```diff
@@ -0,0 +1,262 @@
+/* eslint-disable no-dupe-class-members */
+// Copyright 2017-2019 @polkadot/api authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { AccountId, Address, Call, Extrinsic, ExtrinsicEra, ExtrinsicStatus, EventRecord, Hash, Header, Index } from '@polkadot/types/interfaces';
+import { Callback, Codec, Constructor, IKeyringPair, SignatureOptions } from '@polkadot/types/types';
+import { ApiInterfaceRx, ApiTypes, SignerResult } from '../types';
+import { SignerOptions, SubmittableExtrinsic, SubmittableResultImpl, SubmitableResultResult, SubmitableResultSubscription } from './types';
+
+import { Observable, combineLatest, of } from 'rxjs';
+import { first, map, mergeMap, switchMap, tap } from 'rxjs/operators';
+import { createType, ClassOf, Vec } from '@polkadot/types';
+import { isBn, isFunction, isNumber, isUndefined } from '@polkadot/util';
+
+import { filterEvents, isKeyringPair } from '../util';
+import ApiBase from '../base';
+import SignerPayload from '../SignerPayload';
+import SubmittableResult from './Result';
+
+type Creator<ApiType> = (extrinsic: Call | Uint8Array | string) => SubmittableExtrinsic<ApiType>;
+
+interface SubmittableOptions<ApiType> {
+  api: ApiInterfaceRx;
+  decorateMethod: ApiBase<ApiType>['decorateMethod'];
+  type: ApiTypes;
+}
+
+// The default for 6s allowing for 5min eras. When translating this to faster blocks -
+//   - 4s = (10 / 15) * 5 = 3.33m
+//   - 2s = (10 / 30) * 5 = 1.66m
+const BLOCKTIME = 6;
+const ONE_MINUTE = 60 / BLOCKTIME;
+const DEFAULT_MORTAL_LENGTH = 5 * ONE_MINUTE;
+
+const _Extrinsic: Constructor<Extrinsic> = ClassOf('Extrinsic');
```

Comment:
Need to double-check this - I'm not convinced that in all cases this is lazily created. (The registry certainly tries, however it is not always a simple problem to solve)

If not... and it cannot be convinced to do so... an option is to move the whole `class Submittable ...` into the actual final exported function, so it will only be evaluated when the function is called. (Which is not great since we end up with a monster function there)

Reply:
Yeap, it works. I injected the type `{ Extrinsic: 'ExtrinsicUnknown' }` and as expected it fails.